service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java

```diff
@@ -111,15 +122,54 @@ protected void extractRequiredJsonPayload(final RoutingContext ctx) {
         } else {
             try {
                 if (ctx.getBody() != null) {
-                    ctx.put(KEY_REQUEST_BODY, ctx.getBodyAsJson());
-                    ctx.next();
+                    final var payload = payloadExtractor.apply(ctx);
+                    if (payload != null) {
+                        ctx.put(KEY_REQUEST_BODY, payload);
+                        ctx.next();
+                    } else {
+                        ctx.fail(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, "Null body"));
+                    }
                 } else {
                     ctx.fail(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, "Empty body"));
                 }
             } catch (final DecodeException e) {
-                ctx.fail(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, "Invalid JSON"));
+                ctx.fail(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, "Invalid JSON", e));
             }
         }
+
+    }
+
+    /**
+     * Check the Content-Type of the request to be 'application/json' and extract the payload if this check was
+     * successful.
+     * <p>
+     * The payload is parsed to ensure it is valid JSON and is put to the RoutingContext ctx with the key
+     * {@link #KEY_REQUEST_BODY}.
+     *
+     * @param ctx The routing context to retrieve the JSON request body from.
+     */
+    protected void extractRequiredJsonPayload(final RoutingContext ctx) {
+        extractRequiredJson(ctx, RoutingContext::getBodyAsJson);
+    }
+
+    /**
+     * Check the Content-Type of the request to be 'application/json' and extract the payload if this check was
+     * successful.
+     * <p>
+     * The payload is parsed to ensure it is valid JSON and is put to the RoutingContext ctx with the
+     * key {@link #KEY_REQUEST_BODY}.
+     *
+     * @param ctx The routing context to retrieve the JSON request body from.
+     */
+    protected void extractRequiredJsonArrayPayload(final RoutingContext ctx) {
```

Comment:
is this method supposed to be overrideen by subclasses? If yes, a corresponding comment would be nice, if not, it should be declared final ...

Reply:
This methods signature and concept has been copied over from `extractRequiredJsonPayload`. I can't find either (comment, final) there. What would it be?